src/quic.rs

```diff
@@ -0,0 +1,102 @@
+//! QUIC Transport traits
+//!
+//! This module includes traits and types meant to allow being generic over any
+//! QUIC implementation.
+
+use std::task::{self, Poll};
+
+use bytes::Buf;
+
+// Unresolved questions:
+//
+// - Should the `poll_` methods be `Pin<&mut Self>`?
+
+/// Trait representing a QUIC connection.
+pub trait Connection<B: Buf> {
+    /// The stream type returned when opening a unidirectional send stream.
+    type SendStream: SendStream<B>;
+    /// The stream type returned when accepting a unidirectional receive stream.
+    type RecvStream: RecvStream;
+    /// The stream type returned when accepting or opening a bidirectional
+    /// stream.
+    type BidiStream: SendStream<B> + RecvStream;
+    /// The error type that can be returned when accepting or opening a stream.
+    type Error;
+
+    // Accepting streams
+
+    /// Poll the connection for any received bidirectional streams.
+    fn poll_accept_bidi_stream(
```

Comment:
Maybe we could use an `Option<>` here to signify there won't be any new streams? Or should we assume there won't be any because `Self::Error::ConnectionClosed` or any other error ?

Reply:
Er, yes, you're right! I even had that in the design proposal, but forgot it here.